File: ios/ReservationApp/ReservationApp/Domain/Models/DashboardSummary.swift
//
//  DashboardSummary.swift
//  ReservationApp
//
//  Created by Mehmet Ali Kısacık on 13.11.2025.
//

import Foundation

struct DashboardSummary: Codable {
    let totalReservations: Int
    let totalReservationsChangePercent: Double
    let activeUsers: Int
    let activeUsersChangePercent: Double
    let todayMeals: Int
    let todayMealsChangePercent: Double
    let monthlyCost: Double
    let monthlyCostChangePercent: Double
    // Legacy fields
    let totalUsers: Int
    let totalMeals: Int
    let totalRestaurants: Int
}

File: android/domain/src/main/java/com/reservationapp/domain/model/DashboardSummary.kt
package com.reservationapp.domain.model

data class DashboardSummary(
    val totalReservations: Int,
    val totalReservationsChangePercent: Double,
    val activeUsers: Int,
    val activeUsersChangePercent: Double,
    val todayMeals: Int,
    val todayMealsChangePercent: Double,
    val monthlyCost: Double,
    val monthlyCostChangePercent: Double,
    // Legacy fields
    val totalUsers: Int,
    val totalMeals: Int,
    val totalRestaurants: Int
)

